### PR TITLE
Add minimal Hadoop FileSystem compatibility tester and try it on S3A

### PR DESCRIPTION
**Pull request by arielshaqed**
Add a simple check for a Hadoop FileSystem to be compatible as an underlying FileSystem layer for the upcoming lakeFSFS.  Try it out on S3A (it works!).

### CODE REVIEW

**Review by talSofer — COMMENTED** [review]
LGTM!

**Comment by arielshaqed** [comment]
CLI (that works...):
```sh
AWS_REGION=eu-central-1  AWS_ACCESS_KEY_ID=AKIAforMyS3Account AWS_SECRET_ACCESS_KEY=sssssshhhh/   sbt "~lakefs-spark-examples-247/runMain io.treeverse.clients.examples.S3AUser  s3://treeverse-ariels-test/testing/s3a/"
```
(or replace "247" with "301", of course).

File: clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+          () => compare.download("/abc/g"),
+          () => compare.delete("/abc"),
+          () => failDownload("/abc", fs, mem),
+          () => compare.download("/abc/g"),
+          () => checkObjectStore(mem, objects, prefix)
+        )
+
+        // TODO(ariels): Compare current state on ObjectStore after each action.
+        actions.map(_()).zipWithIndex.filter({ case (res, _) => isNonemptyString(res) }).foreach({ case (err, idx) => {
+          Console.err.println(s"[${idx}] ${err}\n")
```
Comment (nopcoder) [inline clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala:209]: why '\n' when using `println` we already got new line

File: clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    if (args.length != 1) {
+      Console.err.println("Usage: ... s3://bucket/prefix/to/write")
+      System.exit(1)
+    }
+
+    System.setProperty(SDKGlobalConfiguration.ENABLE_S3_SIGV4_SYSTEM_PROPERTY, "true")
+
+    val conf = new Configuration(true)
+    // conf.set("fs.s3a.aws.credentials.provider", """com.amazonaws.auth.profile.ProfileCredentialsProvider""")
+    //    conf.set("fs.s3a.aws.credentials.provider", "org.apache.hadoop.fs.s3a.ProfileCredentialsProvider,org.apache.hadoop.fs.s3a.EnvironmentVariableCredentialsProvider")
```
Comment (nopcoder) [inline clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala:137]: can be removed

**Review by nopcoder — APPROVED** [review]

File: clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala (hunk 2 above)
Comment (arielshaqed) [reply clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala]: Done!

File: clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala (hunk 1 above)
Comment (arielshaqed) [reply clients/spark/examples/src/main/scala/io/treeverse/clients/examples/S3AUser.scala]: OK :-)

**Pull request merged**